account_partner_journal/models/account_move.py

```diff
@@ -0,0 +1,32 @@
+##############################################################################
+#
+#    Copyright (C) 2016-Today: La Louve (<http://www.lalouve.net/>)
+#    @author Julien WESTE
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.move"
+
+    @api.onchange("partner_id", "company_id")
+    def _onchange_partner_id(self):
+        res = super()._onchange_partner_id()
+        if self.partner_id.default_purchase_journal_id:
+            self.journal_id = self.partner_id.default_purchase_journal_id
+        return res
```

Comment:
In Odoo 18, journal_id is a computed field with an inverse_method. https://github.com/odoo/odoo/blob/19.0/addons/account/models/account_move.py#L160

You should not declare an `onchange` method but override the `_search_default_journal` method declared into the base addon https://github.com/odoo/odoo/blob/19.0/addons/account/models/account_move.py#L886

```suggestion
def _search_default_journal(self):
    if self.partner_id.default_purchase_journal_id:
        return self.partner_id.default_purchase_journal_id
    return super()._search_default_journal()
``